src/backend/joanie/core/flows/order.py

```diff
@@ -227,44 +228,24 @@ def failed_payment(self):
         Mark order instance as "failed_payment".
         """
 
-    # ruff: noqa: PLR0911
-    # pylint: disable=too-many-return-statements
     def update(self):
         """
         Update the order state.
         """
-        if self._can_be_state_completed():
-            self.complete()
-            return
-
-        if self.instance.state in [
-            enums.ORDER_STATE_ASSIGNED,
-            enums.ORDER_STATE_TO_SIGN,
-            enums.ORDER_STATE_TO_SAVE_PAYMENT_METHOD,
+        for transition in [
+            self.complete,
+            self.to_sign,
+            self.to_save_payment_method,
+            self.pending,
+            self.pending_payment,
+            self.no_payment,
+            self.failed_payment,
         ]:
-            if self._can_be_state_to_sign():
-                self.to_sign()
-                return
-
-            if self._can_be_state_to_save_payment_method():
-                self.to_save_payment_method()
-                return
-
-            if self._can_be_state_pending():
-                self.pending()
+            try:
+                transition()
                 return
-
-        if self._can_be_state_pending_payment():
-            self.pending_payment()
-            return
-
-        if self._can_be_state_no_payment():
-            self.no_payment()
-            return
-
-        if self._can_be_state_failed_payment():
-            self.failed_payment()
-            return
+            except fsm.TransitionNotAllowed:
+                pass
```

Comment:
Sorry I'm a JS Developer, I like to use more recent things I discover.

```suggestion
with suppress(fsm.TransitionNotAllowed):
    transitino()
```